### PR TITLE
Redis none

### PR DESCRIPTION
**Pull request by chillaq**
# Python SDK

## What did you accomplish?
Added None implementation for Redis
## Extra Notes

### CODE REVIEW

File: splitio/sync/manager.py [hunk 1]
```diff
@@ -127,3 +128,46 @@ def _streaming_feedback_handler(self):
                 self._synchronizer.start_periodic_fetching()
                 _LOGGER.info('non-recoverable error in streaming. switching to polling.')
                 return
+
+class RedisManager(object):  # pylint:disable=too-many-instance-attributes
+    """Manager Class."""
+
+    _CENTINEL_EVENT = object()
```
Comment (mmelograno) [inline splitio/sync/manager.py:135]: why this?

File: splitio/sync/synchronizer.py [hunk 2]
```diff
@@ -377,6 +377,94 @@ def kill_split(self, split_name, default_treatment, change_number):
...
+        :param blocking:flag to wait until tasks are stopped
+        :type blocking: bool
+        """
+        _LOGGER.debug('Shutting down tasks.')
+        self.stop_periodic_data_recording(blocking)
+
+    def start_periodic_data_recording(self):
+        """Start recorders."""
+        _LOGGER.debug('Starting periodic data recording')
+        self._split_tasks.impressions_count_task.start()
```
Comment (mmelograno) [inline splitio/sync/synchronizer.py:414]: wouldn't be possible to count task be none in debug mode?

File: splitio/client/factory.py [hunk 3]
```diff
@@ -424,23 +425,60 @@ def _build_redis_factory(api_key, cfg):
                         _MIN_DEFAULT_DATA_SAMPLING_ALLOWED)
         data_sampling = _MIN_DEFAULT_DATA_SAMPLING_ALLOWED
 
-    imp_strategy = StrategyDebugMode() if cfg['impressionsMode'] == ImpressionsMode.DEBUG else StrategyOptimizedMode(ImpressionsCounter())
+    imp_counter = ImpressionsCounter() if cfg['impressionsMode'] != ImpressionsMode.DEBUG else None
```
Comment (mmelograno) [inline splitio/client/factory.py:428]: move the counter creation to each statement that need it

File: splitio/client/factory.py [hunk 4]
```diff
@@ -424,23 +425,60 @@ def _build_redis_factory(api_key, cfg):
...
         imp_strategy)
 
+    synchronizers = SplitSynchronizers(None, None, None, None,
+        ImpressionsCountSynchronizer(redis_sender_adapter, imp_counter),
+        unique_keys_synchronizer,
+        clear_filter_sync
+    )
+
+    tasks = SplitTasks(None, None, None, None,
+        ImpressionsCountSyncTask(synchronizers.impressions_count_sync.synchronize_counters),
```
Comment (mmelograno) [inline splitio/client/factory.py:457]: impressions count task can be none in debug mode

File: splitio/client/factory.py [hunk 5]
```diff
@@ -424,23 +425,60 @@ def _build_redis_factory(api_key, cfg):
...
+        imp_strategy = StrategyDebugMode()
+    else:
+        imp_strategy = StrategyOptimizedMode(imp_counter)
+
     imp_manager = ImpressionsManager(
         _wrap_impression_listener(cfg['impressionListener'], sdk_metadata),
         imp_strategy)
 
+    synchronizers = SplitSynchronizers(None, None, None, None,
+        ImpressionsCountSynchronizer(redis_sender_adapter, imp_counter),
```
Comment (mmelograno) [inline splitio/client/factory.py:451]: impressions count synchronizer can be none in debug mode

**Review by mmelograno — CHANGES_REQUESTED** [review]

File: splitio/client/factory.py (hunk 5 above)
Comment (chillaq) [reply splitio/client/factory.py:451]: For in memory debug mode, the impressions count synchronizer object is created, do we want to disable it in redis debug mode?

**Pull request merged**